crates/ruma-events/src/sticker.rs

```diff
@@ -6,7 +6,33 @@ use ruma_common::OwnedMxcUri;
 use ruma_macros::EventContent;
 use serde::{Deserialize, Serialize};
 
-use crate::room::ImageInfo;
+#[cfg(not(feature = "compat-encrypted-stickers"))]
+use crate::room::EncryptedFile;
+use crate::room::{ImageInfo, MediaSource};
+
+/// The source of a sticker media file.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[allow(clippy::exhaustive_enums)]
+pub enum StickerMediaSource {
+    /// The MXC URI to the unencrypted media file.
+    #[serde(rename = "url")]
+    Plain(OwnedMxcUri),
+
+    /// The encryption info of the encrypted media file.
+    #[cfg(not(feature = "compat-encrypted-stickers"))]
+    #[serde(rename = "file")]
+    Encrypted(Box<EncryptedFile>),
+}
+
+impl From<StickerMediaSource> for MediaSource {
```

Comment:
This is wrong, it will only work if the feature is enabled so it should be behind the feature flag. Also the opposite conversion might make sense.

Reply:
You didn't handle this comment. Also nitpick, leave an empty line between the impl blocks

Reply:
The From implementation can be used with or without the flag

Reply:
I didn't have any need for it but I can it for completeness